src/support/cond_auto.c

```diff
@@ -55,7 +55,16 @@ __wt_cond_auto_wait_signal(WT_SESSION_IMPL *session, WT_CONDVAR *cond, bool prog
         cond->prev_wait = cond->min_wait;
     else {
         delta = WT_MAX(1, (cond->max_wait - cond->min_wait) / 10);
+        /*
+         * Try to update the previous wait value for the condition variable. There can be multiple
+         * threads doing this concurrently, so use atomic operations to make sure the value remains
+         * within the bounds of the maximum configured. Don't retry if our update didn't make it in
+         * - it's not necessary for the previous wait time to be updated every time.
+         */
+        WT_ORDERED_READ(saved_prev_wait, cond->prev_wait);
         cond->prev_wait = WT_MIN(cond->max_wait, cond->prev_wait + delta);
```

Comment:
We can still race on line 65 and end up with cond->prev_wait more than 1 million, like before the change. In such a case atomic compare on the next line will not fix this value and hence we will get the same bug as before. 

If you remove line 65, I think you fix the bug. order read assures the latest value in the local variable. Atomic compare makes sure that in the case of a race, only the first thread gets to modify successfully, the second thread leaves the value untouched. WT_MIN calculation from the local variable calculates the correct final value.

So i think the change should be :

```
+ WT_ORDERED_READ(saved_prev_wait, cond->prev_wait);
+        __wt_atomic_cas64(
+          &cond->prev_wait, saved_prev_wait, WT_MIN(cond->max_wait, saved_prev_wait + delta));
```

Reply:
+1

Reply:
@agorrod : Yes, I agree with this and discussed it with @sulabhM.  I will do the changes soon.